xdsl/dialects/llvm.py

```diff
@@ -1563,6 +1577,7 @@ def __init__(
         visibility: int | IntegerAttr[IntegerType] = 0,
         sym_visibility: str | StringAttr | None = None,
         body: Region | None = None,
+        other_props: dict[str, Attribute | None] | None = None,
```

Comment:
```suggestion
        other_props: dict[str, Attribute] | None = None,
```
don't see the point in the `None`. it could just be omitted from the dictionary

Reply:
Could make it keyword variadic too if you wanted.

Reply:
I don't have a very strong view on this but it can be quite convenient to not have to do the whole `if thing is not None: props["thing"] = thing`